index.js

```diff
@@ -67,6 +67,15 @@ let cachedRecords;
 let credentials;
 
 /**
+ * Maps between passed in user name and internal user name.
+ *
+ * @type {Object}
+ *
+ * @private
+ */
+let usernameMap;
```

Comment:
I guess it depends on how we look at it, it could be seen as a cached map of usernames/internal usernames since every time we're trying to create a user we'll lookup into it and there's one already there (cached?) we throw an error.